ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -111,860 +107,835 @@
-    final KsqlErrorMessage result = (KsqlErrorMessage) response.getEntity();
-    assertThat(
-        result.getMessage().toLowerCase(),
-        equalTo("incompatible data source type is stream, but statement was drop table"));
-  }
+    // When:
+    final KsqlErrorMessage result = makeFailingRequest(
+        "DROP TABLE test_stream;", Code.BAD_REQUEST);
 
-  @Test
-  public void shouldFailForIncorrectDropTableStatement() {
-    final KsqlResource testResource = TestKsqlResourceUtil.get(ksqlConfig, ksqlEngine);
-    final String ksqlString = "DROP STREAM test_table;";
-    final Response response = handleKsqlStatements(
-        testResource, new KsqlRequest(ksqlString, new HashMap<>()));
-    assertThat(response.getStatus(), equalTo(Response.Status.BAD_REQUEST.getStatusCode()));
-    assertThat(response.getEntity(), instanceOf(KsqlErrorMessage.class));
-    final KsqlErrorMessage result = (KsqlErrorMessage) response.getEntity();
+    // Then:
     assertThat(result.getMessage().toLowerCase(),
-        equalTo("incompatible data source type is table, but statement was drop stream"));
+        is("incompatible data source type is stream, but statement was drop table"));
   }
 
   @Test
-  public void shouldDistributeStatementWithConfig() {
-    final String ksqlString =
-        "CREATE TABLE ORDERS " +
-            "(ORDERTIME BIGINT, ORDERID BIGINT, ITEMID STRING, ORDERUNITS DOUBLE, " +
-            "ARRAYCOL ARRAY<DOUBLE>, MAPCOL MAP<VARCHAR, DOUBLE>) " +
-            "WITH (KAFKA_TOPIC='orders-topic', VALUE_FORMAT='avro', " +
-            "AVRO_SCHEMA_ID='1', KEY='orderid');";
-
-    final CommandId commandId = new CommandId("TABLE", "orders", "CREATE");
-    final QueuedCommandStatus queuedCommandStatus
-        = new QueuedCommandStatus(commandId);
-    final CommandStatus successStatus
-        = new CommandStatus(CommandStatus.Status.SUCCESS, "success");
-    queuedCommandStatus.setFinalStatus(successStatus);
-    final CommandStore commandStore = EasyMock.mock(CommandStore.class);
-    EasyMock.expect(commandStore.enqueueCommand(
-        EasyMock.eq(ksqlString), EasyMock.anyObject(Statement.class),
-        EasyMock.same(ksqlConfig), EasyMock.anyObject(Map.class)))
-        .andReturn(queuedCommandStatus);
-    EasyMock.replay(commandStore);
-
-    final KsqlResource testResource = TestKsqlResourceUtil.get(
-        ksqlConfig, ksqlEngine, commandStore);
-
-    handleKsqlStatements(
-        testResource, new KsqlRequest(ksqlString, new HashMap<>()));
+  public void shouldFailForIncorrectDropTableStatement() {
+    // When:
+    final KsqlErrorMessage result = makeFailingRequest(
+        "DROP STREAM test_table;", Code.BAD_REQUEST);
 
-    EasyMock.verify(commandStore);
+    // Then:
+    assertThat(result.getMessage().toLowerCase(),
+        is("incompatible data source type is table, but statement was drop stream"));
   }
 
-  @SuppressWarnings("unchecked")
   @Test
   public void shouldCreateTableWithInference() {
```

Comment:
If we're collapsing shouldDistributeStatementWithConfig into this test, we should update the name to reflect that.

Reply:
done.